environments/dev-env/CONTRIBUTING.md

```diff
@@ -0,0 +1,510 @@
+
+Create your edge cluster or bring your own k8s edge cluster. In this example, we use [kind](https://kind.sigs.k8s.io/) to create an edge cluster that we name “florin”:
+
+```shell
+kind create cluster --name florin
+```  
+
+### 4. Deploy the kcp-edge platform:
+
+  a) Download kcp binaries for your platform:
+
+  ```shell
+  mkdir kcp
+  cd kcp
+  VERSION=0.11.0 # choose the latest version (without v prefix)
+  OS=darwin   # linux or darwin
+  ARCH=amd64  # arm64 or amd64
+  curl -sSfL "https://github.com/kcp-dev/kcp/releases/download/v${VERSION}/kcp_${VERSION}_${OS}_${ARCH}.tar.gz" > kcp.tar.gz
+  curl -sSfL "https://github.com/kcp-dev/kcp/releases/download/v${VERSION}/kubectl-kcp-plugin_${VERSION}_${OS}_${ARCH}.tar.gz" > kubectl-kcp-plugin.tar.gz
+  ```
+
+  Extract kcp and kubectl-kcp-plugin and place all the files in the bin directories somewhere in your $PATH. For example:
+
+  ```shell
+  tar -xvf kcp.tar.gz
+  tar -xvf kubectl-kcp-plugin.tar.gz
+  export PATH="$PATH:$(pwd)/bin"
+  ```
+
+  b) Start kcp:
+
+  ```shell
+  kcp start >& kcp_log.txt &
+  ```
+  OR
+  
+  If you need to connect to a remote k8s edge cluster, you can bind kcp to the public interface of your virtual machine:
+
+  ```shell
+  kcp start --bind-address <ip-address> >& kcp_log.txt &
+  ```
+  
+  This uses default 6443 port, just bind to a specific address
+  
+  Set your KUBECONFIG variable to the file in `.kcp/admin.kubeconfig` generated by kcp during startup:
+
+  ```shell
+  export KUBECONFIG="$(pwd)/.kcp/admin.kubeconfig"
+  ```
+
+  c) Deploy kcp-edge infra:
+         
+  Run the following command inside the `kcp-edge/environments/dev-env` directory:
+
+  ```bash
+  ./kcp-edge.sh start
+  ```
+
+  The following components are created:
+
+  - 1 kcp workspace: edge service provider workspace (`espw`)
+
+  ```console
+  kubectl ws tree
+  .
+  └── root
+      ├── compute
+      └── espw
+  ```
+
+  - 3 kcp-edge controllers: [edge-scheduler](https://docs.kcp-edge.io/docs/coding-milestones/poc2023q1/edge-scheduler/), [mailbox-controller](https://docs.kcp-edge.io/docs/coding-milestones/poc2023q1/mailbox-controller/) and [placement-translator](https://docs.kcp-edge.io/docs/coding-milestones/poc2023q1/placement-translator/)
+
+  ```shell
+  ps aux | grep -e mailbox-controller -e placement-translator -e scheduler
+  user    11344   0.1  0.1 34906508  28044 s001  S     8:36PM   0:00.43 placement-translator --allclusters-context system:admin -v=2
+  user    11333   0.1  0.1 34885544  19536 s001  S     8:36PM   0:00.38 scheduler -v 2 --root-user kcp-admin --root-cluster root --sysadm-context system:admin --sysadm-user shard-admin
+  user    11323   0.0  0.1 34892244  20780 s001  S     8:36PM   0:00.13 mailbox-controller --inventory-context=root --mbws-context=base -v=2
+  ```
+
+### 5. Connect your edge cluster to the kcp-edge platform:
+
+  a) Create inventory management workspace (`imw`):
+
+  ```console
+  kubectl ws root
```

Comment:
In a "console" block, we show the shell's prompts; this distinguishes input from output.

```console
$ kubectl ws root
Current context is "root"

$ kubectl ws create imw-1 --enter
Current context is "root:imw-1"
```

Reply:
my only reservation on that suggestion - adding a '$' renders the copy button useless - does it not?

Reply:
including the output, which we are not debating, makes the copy button useless

Reply:
We do not own the definition of the "console" tag.

Reply:
Added. I looked at examples in other open source repos and they add the shell's prompts to  distinguishes input from output indeed.